Cython/Utility/ObjectHandling.c

```diff
@@ -1094,12 +1104,23 @@ __Pyx_PEP560_update_bases(PyObject *bases)
             }
             for (j = 0; j < i; j++) {
                 // original code in CPython: base = args[j];
+#if CYTHON_ASSUME_SAFE_MACROS
                 base = PyTuple_GET_ITEM(bases, j);
                 PyList_SET_ITEM(new_bases, j, base);
+#else
+                base = PyTuple_GetItem(bases, j);
+                if (!base) goto error;
+                if (PyList_SetItem(new_bases, j, base) < 0) goto error;
+#endif
                 Py_INCREF(base);
```

Comment:
It's ok to defer the incref in the "safe macros" case, but we shouldn't add a borrowed ref to a list otherwise.
```suggestion
#if CYTHON_ASSUME_SAFE_MACROS
                base = PyTuple_GET_ITEM(bases, j);
                PyList_SET_ITEM(new_bases, j, base);
                Py_INCREF(base);
#else
                base = PyTuple_GetItem(bases, j);
                if (!base) goto error;
                Py_INCREF(base);
                if (PyList_SetItem(new_bases, j, base) < 0) goto error;
#endif
```